src/Features/Core/Portable/SymbolSearch/Windows/SymbolSearchUpdateEngine.Update.cs

```diff
@@ -551,11 +550,26 @@ private async Task<XElement> TryDownloadFileAsync(IFileDownloader fileDownloader
                     XmlResolver = null
                 };
 
-                using var reader = XmlReader.Create(stream, settings);
+                // This code must always succeed.  If it does not, that means that either the server reported bogus data
+                // to the file-downloader, or the file-downloader is serving us bogus data.  In other event, there is
+                // something wrong with those components, and we should both report the issue to Watson, and stop doing
+                // the update.
+                try
+                {
+                    using var reader = XmlReader.Create(stream, settings);
 
-                var result = XElement.Load(reader);
-                await LogInfoAsync("Converting data to XElement completed", cancellationToken).ConfigureAwait(false);
-                return result;
+                    var element = XElement.Load(reader);
+                    await LogInfoAsync("Converting data to XElement completed", cancellationToken).ConfigureAwait(false);
+                    return (element, delay: default);
+                }
+                catch (Exception e) when (_service._reportAndSwallowExceptionUnlessCanceled(e, cancellationToken))
```

Comment:
same q on the `_service` here

Reply:
Answered in other PR.  this allows the tests to actually validate tehse codepaths without actually invoking FatalError (which kills tests).